# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/mtransport/ipc/WebrtcTCPSocketChild.cpp
# Commit: b879a0044cf3
# Full Hash: b879a0044cf39f0230879d075c915485affb7707
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Teach WebrtcTCPSocket to bind to a specific host/port. r=mjf
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45102
# ==============================================================================

diff -r f5353ea9f1df -r b879a0044cf3 media/mtransport/ipc/WebrtcTCPSocketChild.cpp
--- a/media/mtransport/ipc/WebrtcTCPSocketChild.cpp	Tue Sep 17 18:17:14 2019 +0000
+++ b/media/mtransport/ipc/WebrtcTCPSocketChild.cpp	Tue Sep 17 18:17:18 2019 +0000
@@ -64,7 +64,8 @@
 }
 
 void WebrtcTCPSocketChild::AsyncOpen(
-    const nsCString& aHost, const int& aPort, const net::LoadInfoArgs& aArgs,
+    const nsCString& aHost, const int& aPort, const nsCString& aLocalAddress,
+    const int& aLocalPort, const net::LoadInfoArgs& aArgs,
     const nsCString& aAlpn, const dom::TabId& aTabId,
     NrSocketProxyConfig::ProxyPolicy aProxyPolicy) {
   LOG(("WebrtcTCPSocketChild::AsyncOpen %p %s:%d\n", this, aHost.get(), aPort));
@@ -85,7 +86,8 @@
                                                                         aTabId);
   }
 
-  SendAsyncOpen(aHost, aPort, aArgs, aAlpn, aProxyPolicy);
+  SendAsyncOpen(aHost, aPort, aLocalAddress, aLocalPort, aArgs, aAlpn,
+                aProxyPolicy);
 }
 
 }  // namespace net